framework/include/userobjects/ElementSubdomainModifier.h

```diff
@@ -97,32 +113,60 @@ class ElementSubdomainModifier : public ElementUserObject
   /// Change the subdomain ID of all ancestor elements
   void setAncestorsSubdomainIDs(const SubdomainID & subdomain_id, const dof_id_type & elem_id);
 
-  /// Helper function to build the range of moved elements
-  void buildMovedElemsRange();
+  /// Helper function to build the range of activated elements
+  void buildActivatedElemsRange();
+
+  /// Helper function to build the range of activated nodes
+  void buildActivatedNodesRange();
 
-  /// Helper function to build the range of boundary nodes on moved elements
-  void buildMovedBndNodesRange();
+  /// Helper function to build the range of activated boundary nodes
+  void buildActivatedBndNodesRange();
 
-  /// Set old and older solutions to be the same as the current solution
-  void setOldAndOlderSolutionsForMovedNodes(SystemBase & sys);
+  /// Find nearest dofs for each variable
+  void findNearestDofs(SystemBase &, const VariableName &);
+
+  /// Set current solution to be the same as the nearest dof's old value
+  void
+  setNearestSolutionForActivatedDofs(SystemBase &, const VariableName &, NumericVector<Real> &);
+
+  /// Set activated dofs to constant
+  void setConstantForActivatedDofs(SystemBase & sys, const VariableName & var_name);
 
   /// Elements on the undisplaced mesh whose subdomain IDs have changed
   std::vector<const Elem *> _moved_elems;
 
   /// Elements on the displaced mesh whose subdomain IDs have changed
   std::vector<const Elem *> _moved_displaced_elems;
 
-  /// Nodes on the moved elements
-  std::set<dof_id_type> _moved_nodes;
+  /// Newly activated elements
+  std::vector<const Elem *> _activated_elems;
```

Comment:
Since we are doing std::find() possibly frequently, maybe it will be more efficient to use std::unordered_set<> here?